File: frontend/src/types/invoice.ts
export type InvoiceLineForm = {
  rowId: string;
  itemCode: string;
  itemId?: number;
  itemName?: string;
  quantity: number;
  price?: number;
  subtotal?: number;
};

export type ClientForm = {
  senderName: string;
  senderAddress: string;
  receiverName: string;
  receiverAddress: string;
};

export type WizardStep = 1 | 2 | 3;

export type WizardState = {
  step: WizardStep;
  client: ClientForm;
  items: InvoiceLineForm[];
};

export type SubmitInvoicePayload = {
  sender_name: string;
  sender_address: string;
  receiver_name: string;
  receiver_address: string;
  items: Array<{
    item_id?: number;
    code: string;
    quantity: number;
    price?: number;
    subtotal?: number;
  }>;
};

export type CreatedInvoiceResponse = {
  id: number;
  invoice_number: string;
  sender_name: string;
  sender_address: string;
  receiver_name: string;
  receiver_address: string;
  total_amount: number;
  created_by: number;
  created_at: string;
  details: Array<{
    id: number;
    invoice_id: number;
    item_id: number;
    quantity: number;
    price: number;
    subtotal: number;
    item: {
      id: number;
      code: string;
      name: string;
      price: number;
    };
  }>;
};

File: backend/internal/dto/invoice.go
package dto

type CreateInvoiceItemRequest struct {
	ItemID   *uint  `json:"item_id"`
	Code     string `json:"code"`
	Quantity int    `json:"quantity"`
	Price    *int64 `json:"price,omitempty"`
	Subtotal *int64 `json:"subtotal,omitempty"`
}

type CreateInvoiceRequest struct {
	SenderName      string                     `json:"sender_name"`
	SenderAddress   string                     `json:"sender_address"`
	ReceiverName    string                     `json:"receiver_name"`
	ReceiverAddress string                     `json:"receiver_address"`
	Items           []CreateInvoiceItemRequest `json:"items"`
}